node/tests/redis-modules/RedisModules.test.ts

```diff
@@ -0,0 +1,348 @@
+
+        it.each([
+            [true, ProtocolVersion.RESP2],
+            [true, ProtocolVersion.RESP3],
+            [false, ProtocolVersion.RESP2],
+            [false, ProtocolVersion.RESP3],
+        ])(
+            `test json set conditional set_%p%p`,
+            async (clusterMode, protocol) => {
+                await runTest(
+                    async (client: BaseClient) => {
+                        const key = uuidv4();
+                        const json_value = { a: 1.0, b: 2 };
+                        expect(
+                            await json.set(
+                                client,
+                                key,
+                                "$",
+                                JSON.stringify(json_value),
+                                ConditionalChange.OnlyIfExist,
+                            ),
+                        ).toBeNull();
+
+                        expect(
+                            await json.set(
+                                client,
+                                key,
+                                "$",
+                                JSON.stringify(json_value),
+                                ConditionalChange.OnlyIfDoesNotExist,
+                            ),
+                        ).toBe("OK");
+
+                        expect(
+                            await json.set(
+                                client,
+                                key,
+                                "$.a",
+                                "4.5",
+                                ConditionalChange.OnlyIfDoesNotExist,
+                            ),
+                        ).toBeNull();
+
+                        const result = await json.get(client, key, "$.a");
+                        expect(JSON.parse(result!)).toEqual([1.0]);
+
+                        expect(
+                            await json.set(
+                                client,
+                                key,
+                                "$.a",
+                                "4.5",
+                                ConditionalChange.OnlyIfExist,
+                            ),
+                        ).toBe("OK");
+
+                        expect(await json.get(client, key, ".a")).toBe("4.5");
+                    },
+                    clusterMode,
+                    protocol,
+                );
+            },
+            config.timeout,
+        );
+
+        it.each([
+            [true, ProtocolVersion.RESP2],
+            [true, ProtocolVersion.RESP3],
+            [false, ProtocolVersion.RESP2],
+            [false, ProtocolVersion.RESP3],
+        ])(
+            `Test JSON get formatting, cluster mode: %s, and protocol: %s`,
+            async (clusterMode, protocol) => {
+                await runTest(
+                    async (client: BaseClient) => {
+                        const key = uuidv4();
+                        const json_value = { a: 1.5, b: 2, c: { d: 3, e: 4 } };
```

Comment:
Can you have a separate test checking that the client works with all data types? (int, float, bool, string..)